lib/install.js

```diff
@@ -382,11 +383,13 @@ Installer.prototype.loadAllDepsIntoIdealTree = function (cb) {
     steps.push([validateArgs, this.idealTree, this.args])
     steps.push([loadRequestedDeps, this.args, this.idealTree, saveDeps, cg.newGroup('loadRequestedDeps')])
   } else {
-    if (this.prod) {
+    if (this.prod && this.dev) {
```

Comment:
Interesting.

I initially made this change, because the other fix (considering dev dependencies) caused the `shrinkwrap-dev-depdency.js` test to fail.  

In that scenario, package.json asks for this:

```
pkg@1.0.0 
├── DEV: depA@1.0.0
├─┬ libA@1.0.0
│ ├── depA@1.0.0
```

The test case expects us to install this:
```
pkg@1.0.0 
├── depA@1.0.0
├── libA@1.0.0
```

What we actually get (without my `loadAllDeps` change) is this:
```
pkg@1.0.0 
├── depA@1.0.0
├─┬ libA@1.0.0
│ ├── depA@1.0.0
```

Unlike your example (which is indeed valid), this tree is needlessly suboptimal.  Nobody benefits from this behavior -- the tree is bigger, and looks _less_ like the prod-only tree.  

Curiously, if `loadDevDeps` is run separately, but before `loadDeps`, we get the correct tree.

This seems to relate to our discussion about `earliestInstallable` above.  When processing devDeps, `findRequirement` doesn't seem to have picked up on the copy of `depA` in the root folder.  `earliestInstallable` sees a copy of `depA` in the root folder, doesn't realize that it's a compatible version (because it trusted that `findRequrement` to have previously found that for us), and decides to hoist a new copy of `depA` into `libA`.